Core/Src/main.c

```diff
@@ -9,46 +9,57 @@
  * 
  */
 
-#include <string.h>
 #include <stdio.h>
-
-#include "stm32f0xx.h"
-#include "stm32f051_rcc.h"
-#include "stm32f051_gpio.h"
-#include "stm32f051_uart.h"
-#include "stm32f051_i2c.h"
+#include <stdlib.h>
+#include <string.h>
 
 #include "main.h"
 #include "pins.h"
 
 #include "ds3231.h"
+#include "uart_ring.h"
+
+struct
+{
+  char strOut[100];
+  uint8_t count;
+  char bufIn[100];
+} bufUART;
 
-uint8_t bufUART[256] = {};
 DS3231_t DS3231;
+UART_Ring_t uart1;
+STM32F051_UART_t* puart1 = &uart1.uart;
 
-char commandFromUART = 0;
+uint16_t delay = 500;
 
 void UARTComander(void)
 {
-  if(commandFromUART == 'd')
+  if(strcmp(bufUART.bufIn, "Set date") == 0)
   {
-    DS3231_Date_t date;
-    date.Date = 17;
-    date.Month = 1;
-    date.Year = 21;
-    date.Day = 7;
+    DS3231.Date.Date = 18;
+    DS3231.Date.Month = 1;
+    DS3231.Date.Year = 21;
+    DS3231.Date.Day = 1;
 
-    DS3231_SetDate(&DS3231, &date);
+    DS3231_SetDate(&DS3231, &DS3231.Date);
   }
-  else if (commandFromUART == 't')
+  else if (strcmp(bufUART.bufIn, "Set time") == 0)
   {
-    DS3231_Time_t time;
-    time.Hours = 14;
-    time.Minutes = 9;
-    time.Seconds = 20;
+    DS3231.Time.Hours = 4;
+    DS3231.Time.Minutes = 47;
+    DS3231.Time.Seconds = 20;
 
-    DS3231_SetTime(&DS3231, &time);
+    DS3231_SetTime(&DS3231, &DS3231.Time);
   }
+  else {
```

Comment:
Don't forget about the code style we have chosen.
`{` must be placed at the beginning of the next line